atlp-rwanda/knights-ecomm-fe
# #16 Implement Login Functionality

#### What does this PR do?
This PR enhances the user login functionality by adding options for users to sign in with email/password and Google.

#### Description of Task to be completed?
- Implement UI design for login and ensure its responsiveness.
- Integrated login API and google authentication API on the frontend
- Implemented Redux for state management of login process and user sessions.
- Added error messages for incorrect login details

#### How should this be manually tested?
- Ensure You have backend service running and set VITE_APP_API_URL in your .env file.
- Clone this repository and navigate to the project directory
  ```
  git clone -b ft-register https://github.com/atlp-rwanda/knights-ecomm-fe.git
  cd knights-ecomm-fe
  git checkout ft-login
  npm i
  npm run dev
  ```

#### What are the relevant pivotal tracker/Trello stories?
#16 

#### Screenshots
![Screenshot from 2024-06-21 16-48-06](https://github.com/atlp-rwanda/knights-ecomm-fe/assets/82057476/2003fac6-5018-4ff6-a05c-fe422caf76c8)
